plugins/ftdidmx/FtdiWidget.h

```diff
@@ -48,140 +47,186 @@ namespace plugin {
 namespace ftdidmx {
 
 /**
- * This class holds information about an attached ftdi chip
+ * This class holds information about an attached ftdi chips
  */
 class FtdiWidgetInfo {
  public:
-    FtdiWidgetInfo(const std::string &name,
-                   const std::string &serial,
-                   unsigned int id)
-      : m_name(name),
-        m_serial(serial),
-        m_id(id) {
-    }
-
-    FtdiWidgetInfo(const FtdiWidgetInfo &info)
-      : m_name(info.Name()),
-        m_serial(info.Serial()),
-        m_id(info.Id()) {
-    }
-
-    virtual ~FtdiWidgetInfo() {}
-
-    std::string Name() const { return m_name; }
-    std::string Serial() const { return m_serial; }
-    unsigned int Id() const { return m_id; }
-
-    std::string Description() const {
-      return m_name + " with serial number : " + m_serial + " ";
-    }
-
-    FtdiWidgetInfo& operator=(const FtdiWidgetInfo &other) {
-      if (this != &other) {
-        m_name = other.Name();
-        m_serial = other.Serial();
-        m_id = other.Id();
-      }
-      return *this;
+  static const int ftdi_vid = 0x0403;
+  static const int ft232_pid = 0x6001;
+
+  FtdiWidgetInfo(const std::string &name,
+                  const std::string &serial,
+                  unsigned int id,
+                  const uint16_t vid = ftdi_vid,
+                  const uint16_t pid = ft232_pid
+                )
+    : m_name(name),
+      m_serial(serial),
+      m_id(id),
+      m_vid(vid),
+      m_pid(pid) {
+  }
+
+  FtdiWidgetInfo(const FtdiWidgetInfo &info)
+    : m_name(info.Name()),
+      m_serial(info.Serial()),
+      m_id(info.Id()),
+      m_vid(info.Vid()),
+      m_pid(info.Pid()) {
+  }
+
+  ~FtdiWidgetInfo() {}
+
+  std::string Name() const { return m_name; }
+  std::string Serial() const { return m_serial; }
+
+  unsigned int Id() const { return m_id; }
+  uint16_t Vid() const { return m_vid; }
+  uint16_t Pid() const { return m_pid; }
+
+  std::string Description() const {
+    return m_name + " with serial number : " + m_serial + " ";
+  }
+
+  FtdiWidgetInfo& operator=(const FtdiWidgetInfo &other) {
+    if (this != &other) {
+      m_name = other.Name();
+      m_serial = other.Serial();
+      m_id = other.Id();
+      m_vid = other.Vid();
+      m_pid = other.Pid();
     }
+    return *this;
+  }
 
  private:
-    std::string m_name;
-    std::string m_serial;
-    unsigned int m_id;
-};
+  std::string m_name;
+  std::string m_serial;
 
+  unsigned int m_id;
+  uint16_t m_vid;
+  uint16_t m_pid;
+};
 
 /**
  * An FTDI widget
  */
 class FtdiWidget {
```

Comment:
It looks like FtdiWidget and FtdiWidgetInfo are almost identical. Can they be collapsed into a single class?